### PR TITLE
Add portaudio module

### PR DESCRIPTION
**Pull request by bramtayl**
Using imported targets has several benefits:

- Packaging include-dirs and libraries together for cleaner code
- Automatic handling of usage requirements (see https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#target-usage-requirements)
- Tracking of runtime dependencies (https://cmake.org/cmake/help/latest/command/install.html#runtime-dependency-set)
- Unifies vcpkg and non-vcpkg builds

### CODE REVIEW

File: platform/vcpkg/ports/portaudio/vcpkg.json
Comment (bramtayl) [inline platform/vcpkg/ports/portaudio/vcpkg.json:1]: @stekyne was there are reason you had a custom PortAudio port? Looks similar to the vcpkg default

Comment (stekyne) [reply platform/vcpkg/ports/portaudio/vcpkg.json:1]: Yeah, there is a missing component that has to be downloaded so I had to fork it for Csound to get the same version of Portaudio we had before VCPKG.

File: InOut/CMakeLists.txt [hunk 1]
```diff
@@ -113,15 +109,14 @@ if(USE_PULSEAUDIO)
     make_plugin(rtpulse rtpulse.c ${PULSEAUDIO_LIBRARY} ${PULSESIMPLE_LIBRARY})
 endif()
 
-if(PORTAUDIO_FOUND OR USE_VCPKG)
+if(USE_PORTAUDIO AND ((TARGET portaudio_static) OR (TARGET portaudio)))
```
Comment (stekyne) [inline InOut/CMakeLists.txt:112]: Is this the recommended way to check if a target exists? Isn't it convention to set `HAVE_X` package?

Comment (bramtayl) [reply InOut/CMakeLists.txt:112]: Yeah, but the vcpkg portaudio find module doesn't set a variable like that

File: platform/vcpkg/ports/portaudio/vcpkg.json
Comment (bramtayl) [reply platform/vcpkg/ports/portaudio/vcpkg.json:1]: Oh I see. Ok, ref #1765

File: InOut/CMakeLists.txt (hunk 1 above)
Comment (stekyne) [reply InOut/CMakeLists.txt:112]: Yeah but if you have `find_package(portaudio CONFIG REQUIRED)` the whole build fails if it's not found. It seems redundant to check here in this kind of awkward way.

Comment (bramtayl) [reply InOut/CMakeLists.txt:112]: > find_package(portaudio CONFIG REQUIRED) the whole build fails if it's not found

That would make sense to me too. But @kunstmusik was saying that we want to turn variables like `USE_PORTAUDIO` off if the dependencies aren't found. I would have done this with `check_deps`, but I couldn't figure out how to get the `(TARGET portaudio)` conditions to pass through as function arguments.

Comment (bramtayl) [reply InOut/CMakeLists.txt]: Added `check_deps` for clarity

Comment (stekyne) [reply InOut/CMakeLists.txt]: Ah right. Yeah I see what you mean. There is a flag `FAIL_NOT_FOUND` or something that we use to fail the build if a dependency is missing. We could possibly use it in that case.  Maybe we will need to explicitly check for the targets so

**Comment by bramtayl** [comment]
Bump?

**Comment by stekyne** [comment]
What is this needed for? It's been said elsewhere, you need to explain the changes and the justification for making alterations to something that has already been working. What's the advantage of it?

**Comment by bramtayl** [comment]
Using imported targets has a few benefits:

- Packaging include-dirs and libraries together for cleaner code
- Automatic handling of usage requirements (see https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#target-usage-requirements)
- Tracking of runtime dependencies (https://cmake.org/cmake/help/latest/command/install.html#runtime-dependency-set)
- Unify vcpkg and non-vcpkg builds

**Comment by bramtayl** [comment]
Part of #1811 now

**Pull request closed without merging**